### PR TITLE
UF-11102: Implemented logic for applying custom sortorder on checklists

### PR DESCRIPTION
**Pull request by josc04**
- Implemented logic for applying custom sortorder on checklist templates and employee checklists
- Added organizational starting point for sorting to organization API to be able to apply closest sort order based on provided input
- Adjusted method for saving mentor on employee checklist to not return checklist to have same return type as when deleting mentor
- Small additions and fixes
- Extended organization IT-tests to verify custom sort order functionality
- Etended employee checklist IT-tests to verify custom sort order functionality

## Types of changes

- [ ] Bug fix
- [x] New feature
- [x] Removed feature
- [ ] Code style update (formatting etc.)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Content/Data

## Does this PR introduce a breaking change?
- [x] Yes (Adjustments made in current version as service is new and have no clients)
- [ ] No
      
## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly (if applicable).
- [x] I have added/updated tests to cover my changes (if applicable).

### CODE REVIEW

File: src/main/java/se/sundsvall/checklist/service/ChecklistService.java [hunk 1]
```diff
@@ -122,6 +127,9 @@ public void deleteChecklist(final String municipalityId, final String id) {
 			organizationRepository.save(organization);
 		});
 
+		// Remove custom sort options for components in the checcklist
```
Comment (mha109) [inline src/main/java/se/sundsvall/checklist/service/ChecklistService.java:130]: checcklist ;)

File: src/main/java/se/sundsvall/checklist/service/util/SortingUtils.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	}
+
+	/**
+	 * Helper method for arranging phases and their respective tasks in the correct order for employee checklists
+	 *
+	 * @param  phases the list of phases to sort
+	 * @return        the sorted list
+	 */
+	public static List<Phase> sortPhases(final List<Phase> phases) {
+		ofNullable(phases).orElse(emptyList()).stream()
```
Comment (mha109) [inline src/main/java/se/sundsvall/checklist/service/util/SortingUtils.java:59]: is `stream()` needed here?

File: src/main/java/se/sundsvall/checklist/service/util/SortingUtils.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+	}
+
+	/**
+	 * Helper method for arranging phases and their respective tasks in the correct order for employee checklists
+	 *
+	 * @param  phases the list of phases to sort
+	 * @return        the sorted list
+	 */
+	public static List<EmployeeChecklistPhase> sortEmployeeChecklistPhases(final List<EmployeeChecklistPhase> phases) {
+		ofNullable(phases).orElse(emptyList()).stream()
```
Comment (mha109) [inline src/main/java/se/sundsvall/checklist/service/util/SortingUtils.java:121]: and an extra `.stream()` here as well? :)

File: src/test/java/se/sundsvall/checklist/service/EmployeeChecklistServiceTest.java
Comment (mha109) [inline src/test/java/se/sundsvall/checklist/service/EmployeeChecklistServiceTest.java:1]: missing `verifyNoMoreInteractions()`

File: src/main/java/se/sundsvall/checklist/service/util/SortingUtils.java (hunk 2 above)
Comment (josc04) [reply src/main/java/se/sundsvall/checklist/service/util/SortingUtils.java:59]: Nope

**Review by mha109 — APPROVED** [review]

**Pull request merged**